File: old/rin.Framework/Widgets/Events/CharacterEvent.cs
using rin.Framework.Graphics.Windows;
using rin.Framework.Widgets.Graphics;

namespace rin.Framework.Widgets.Events;

public class CharacterEvent(Surface surface, char character,InputModifier mods) : Event(surface)
{
    public char Character = character;
    public InputModifier Mods = mods;
}

File: lib/rin/gui/events/CharacterEvent.cpp
#include "rin/gui/events/CharacterEvent.h"
namespace rin::gui
{
    CharacterEvent::CharacterEvent(Surface* inSurface, const char& inCharacter, const Flags<io::InputModifier>& inModifiers) : Event(inSurface), character(inCharacter), modifiers(inModifiers)
    {
    }
}
